src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHouseLookupDirectService.cs

```diff
@@ -20,9 +20,34 @@ public CompaniesHouseLookupDirectService(HttpClient httpClient)
     {
         var response = await _httpClient.GetAsync($"{CompaniesHouseEndpoint}/{id}");
 
-        if (response.StatusCode == HttpStatusCode.NoContent || response.StatusCode == HttpStatusCode.NotFound || response.StatusCode == HttpStatusCode.InternalServerError)
+        if (response.StatusCode == HttpStatusCode.NoContent || response.StatusCode == HttpStatusCode.NotFound)
         {
-            return null;
+            return new Company
+            {
+               Error = new UploadFileErrorModel
+                {
+                   FileLineNumber = 1,
+                   FileContent = string.Empty,
+                   Message = BulkUpdateErrors.ResourceNotFoundErrorMessage,
+                   ErrorNumber = BulkUpdateErrors.ResourceNotFoundError,
+                   IsError = true
+                }
+            };
+        }
+
+        if (response.StatusCode == HttpStatusCode.InternalServerError)
+        {
+            return new Company
+            {
+               Error = new UploadFileErrorModel
+               {
+                   FileLineNumber = 1,
+                   FileContent = string.Empty,
+                   Message = BulkUpdateErrors.ResourceNotReachableErrorMessage,
+                   ErrorNumber = BulkUpdateErrors.ResourceNotReachableError,
+                   IsError = true
+               }
+            };
```

Comment:
Line 53 will throw an exception and the process will stop. You should check for IsSuccessStatusCode and set error instead of allowing it to throw an exception

Reply:
changed as discussed.